.github/workflows/storybook.yml

```diff
@@ -8,9 +20,33 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
         with:
-          node-version: "16"
+          node-version: "18"
           cache: "npm"
+          
+      - name: Setup Pages
+        uses: actions/configure-pages@v3
+
       - name: install dependencies
         run: npm ci
-      # - name: run build
-      #   run: npm run build-storybook
+      - name: run build
+        run: npm run build-storybook
+
+      - name: Upload artifact
+        if: github.event_name == 'push'
+        uses: actions/upload-pages-artifact@v1
+        with:
+          path: './storybook-static'
+
+# Deployment job
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'
```

Comment:
I think this will run for every push, where I think you wanted it just for changes in `main`? Maybe something like this `refs/heads/main`?

Reply:
the `on` settings are made so it will only run on main.
```yml
push:
    branches:
      - main
    paths:
      - pages/**
      - components/**
      - stories/**
```

Reply:
But it has `pull_request` on there also? https://github.com/EddieHubCommunity/LinkFree/pull/3632/files#diff-97cca8fb169bc3951469886bf2f73b7bcc362a6b6da11325c7a617f69a797248R10

Reply:
I am just thinking for `pull_request` it should only build and not deploy?